tests/test_integration.py

```diff
@@ -393,49 +446,47 @@ def _get_mem_vm():
         # Online
         self.assertEqual(_get_mem_hv(), 2048)
         vm_mem = _get_mem_vm()
-        mem_set(self.vm.server_obj['hostname'], '+1G')
-        self.vm.reload()
+        mem_set(self.vm_obj['hostname'], '+1G')
+        self.vm_obj = self.get_vm_obj()
         self.assertEqual(_get_mem_hv(), 3072)
         self.assertEqual(_get_mem_vm() - vm_mem, 1024)
 
         with self.assertRaises(Warning):
-            mem_set(self.vm.server_obj['hostname'], '3G')
+            mem_set(self.vm_obj['hostname'], '3G')
 
         with self.assertRaises(InvalidStateError):
-            mem_set(self.vm.server_obj['hostname'], '2G')
+            mem_set(self.vm_obj['hostname'], '2G')
 
         with self.assertRaises(IGVMError):
-            mem_set(self.vm.server_obj['hostname'], '200G')
+            mem_set(self.vm_obj['hostname'], '200G')
 
         # Not dividable
         with self.assertRaises(IGVMError):
-            mem_set(self.vm.server_obj['hostname'], '4097M')
+            mem_set(self.vm_obj['hostname'], '4097M')
 
-        self.vm.reload()
+        self.vm_obj = self.get_vm_obj()
         self.assertEqual(_get_mem_hv(), 3072)
         vm_mem = _get_mem_vm()
         self.vm.shutdown()
 
         with self.assertRaises(IGVMError):
-            mem_set(self.vm.server_obj['hostname'], '200G')
+            mem_set(self.vm_obj['hostname'], '200G')
 
-        mem_set(self.vm.server_obj['hostname'], '1024M')
-        self.vm.reload()
+        mem_set(self.vm_obj['hostname'], '1024M')
+        self.vm_obj = self.get_vm_obj()
         self.assertEqual(_get_mem_hv(), 1024)
 
-        mem_set(self.vm.server_obj['hostname'], '2G')
-        self.vm.reload()
+        mem_set(self.vm_obj['hostname'], '2G')
+        self.vm_obj = self.get_vm_obj()
         self.assertEqual(_get_mem_hv(), 2048)
         self.vm.start()
         self.assertEqual(_get_mem_vm() - vm_mem, -1024)
 
     def test_vcpu_set(self):
-        buildvm(self.vm.server_obj['hostname'])
-
         def _get_hv():
             # Xen does not provide values when VM is powered off
```

Comment:
Outdated comment.

Reply:
Cleaning XEN things was not scope of my work. We probably can even remove the code below, but let's do it in separate merge request.